storage/block/result.go

```diff
@@ -88,10 +98,78 @@ type fetchBlocksMetadataResult struct {
 	blocks []FetchBlockMetadataResult
 }
 
-// newFetchBlocksMetadataResult creates new database blocks metadata
-func newFetchBlocksMetadataResult(id string, blocks []FetchBlockMetadataResult) FetchBlocksMetadataResult {
+// NewFetchBlocksMetadataResult creates new database blocks metadata
+func NewFetchBlocksMetadataResult(id string, blocks []FetchBlockMetadataResult) FetchBlocksMetadataResult {
 	return fetchBlocksMetadataResult{id: id, blocks: blocks}
 }
 
 func (m fetchBlocksMetadataResult) ID() string                         { return m.id }
 func (m fetchBlocksMetadataResult) Blocks() []FetchBlockMetadataResult { return m.blocks }
+
+type filteredBlocksMetadataIter struct {
+	start     time.Time
+	end       time.Time
+	blockSize time.Duration
+	res       []FetchBlocksMetadataResult
+
+	id       string
+	metadata Metadata
+	ri       int
```

Comment:
`resultIdx` and `blockIdx` as you mentioned. Not as terse as `ri` and `bi` though but yeah can change